src/utils/sendDiscordDm.ts

```diff
@@ -0,0 +1,48 @@
+// TO send discord DM we need 2 APIs.
+// 1. create a channel for dm
+// 2. send the message over the created channel.
+
+import { DISCORD_BASE_URL } from "../constants/urls";
+import { env } from "../typeDefinitions/default.types";
+import { createDmChannel } from "../typeDefinitions/discordMessage.types";
+import { discordTextResponse } from "./discordResponse";
+
+/**
+ *
+ * @param userId {number}: user id of the person using the slash command received from the interaction
+ * @param env {env}: contains environment variables
+ */
+
+export const sendDiscordDm = async (userId: number, env: env) => {
+  // "/users/@me/channels" is an endpoint provided by discord to create a dm channel
+  const createDmChannel: createDmChannel = await fetch(
+    `${DISCORD_BASE_URL}/users/@me/channels`,
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+      body: JSON.stringify({
+        content: discordTextResponse("Hello"),
+        recipient_id: userId,
+      }),
+    }
+  ).then((res) => {
+    return res.json();
+  });
+
+  const channelId = createDmChannel.id;
+
+  // "/channels/{channel.id}/messages" is used to send a message to the specified channel
+  await fetch(`${DISCORD_BASE_URL}/channels/${channelId}/messages`, {
```

Comment:
we should wrap this up in a `try - catch` block

Reply:
thanks for the comment. I added it.